scripts/release-notes-jira.mjs

```diff
@@ -0,0 +1,113 @@
+/**
+ * Generates release notes from completed Jira stories and bugs.
+ * 
+ * Required command-line parameters:
+ * - Jira project key (e.g., "LARA")
+ * - Jira label for filtering stories (e.g., "lara-5.0.0")
+ * 
+ * Optional command-line parameters:
+ * - "slack" to format output for Slack
+ * 
+ * Example usage:
+ * node release-notes-jira.mjs LARA lara-5.0.0 slack
+ */
+
+import "dotenv/config"
+import fetch from "node-fetch";
+import querystring from "querystring";
+import { extractBlurbText, jiraApiBaseUrl, jiraRequestHeaders } from "./utils.mjs";
+
+const jiraUser = process.env.JIRA_USER;
+const jiraToken = process.env.JIRA_TOKEN;
+
+if (!jiraUser || !jiraToken) {
+  console.error("Both the JIRA_USER and JIRA_TOKEN environment variables are required.");
+  process.exit(1);
+}
+
+const features = [];
+const bugs = [];
+const underTheHood = [];
+
+function isUnderTheHood(story) {
+  return story.fields.labels.find(label => label === "under-the-hood");
+}
+
+async function collectStories(projectKey, search) {
+  const fields = ["id", "summary", "issuetype", "description", "labels"].join(",");
+  const jql = `project=${projectKey} AND issuetype in (Story, Bug) AND status in (Done, Closed) AND labels in (${search})`;
+  const urlQuery = querystring.stringify({
+    jql,
+    fields,
+    maxResults: 100
+  });
+  const url = `${jiraApiBaseUrl}/search?${urlQuery}`;
+  const requestHeaders = jiraRequestHeaders(jiraUser, jiraToken);
+  const response = await fetch(url, requestHeaders);
+  const json = await response.json();
+  const stories = json.issues;
+
+  if (stories.length === 0) {
+    console.error(`No stories found for project ${projectKey} with label ${search}`);
+    process.exit(1);
+  }
+
+  for (const story of stories) {
+    if (isUnderTheHood(story)) {
+      underTheHood.push(story);
+    } else {
+      if (story.fields.issuetype.name === "Story") {
+        features.push(story);
+      }
+      if (story.fields.issuetype.name === "Bug") {
+        bugs.push(story);
+      }  
+    }
+  }
+}
+
+const jiraProjectKey = process.argv[2];
+const jiraLabel = process.argv[3];
+const slack = process.argv?.[4];
+await collectStories(jiraProjectKey, jiraLabel);
+
+function storyText(story) {
+  const blurbText = extractBlurbText(story.fields.description.content);
+  if (blurbText) {
+    return blurbText;
+  }
+  return story.fields.summary.trim();
+}
+
+function storyItem(story) {
+  const text = storyText(story);
+  return slack 
+    ? `*[${story.key}](${jiraApiBaseUrl}/browse/${story.key}):* ${text}` 
+    : `**${story.key}:** ${text}`;
```

Comment:
Unless the Jira github app does it automatically, we'll have to add a setting to each GitHub project so that it automatically converts strings that match a jira issue ids to links. This has been setup for many repositories to support a pattern of `PT-[number]`. So we'd need to change it to be support `[Jira project code]-[number]`. That will be necessary for non-slack release notes here to work property.

Slack can basically do the same thing if the jira project is added to the channel where the release notes are being sent. So that might be worth doing and then this slack version could be simplified here to just have 1 `*` instead of 2 `*`.